### PR TITLE
feat: fix core-api sentry

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
- 完善core-api sentry及日志sentry
- 新增gin recover sentry插件

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/core-api/cmd/init.go [hunk 1]
```diff
@@ -76,11 +77,17 @@ func initSentry() {
 			return
 		}
```
Comment (wklken) [inline src/core-api/cmd/init.go:78]: 改下判断, 去掉 Sentry.Enable字段, 根据dsn是否为空确定是否初始化

File: src/core-api/pkg/logging/init.go [hunk 2]
```diff
@@ -49,15 +52,41 @@ func newZapSugarLogger() *zap.SugaredLogger {
...
-		apiLogger = newZapJSONLogger(&logger.API)
+		apiLogger = newZapJSONLogger(&config.Logger.API, options)
 	})
 }
 
+// newSentryLogCore
+func newSentryLogCore(cfg *config.Config) (zapcore.Core, error) {
+	rawCore := zapsentry.NewCore(zapcore.Level(cfg.Sentry.ReportLogLevel), sentry.CurrentHub().Client())
+	sentryCore := zapcore.RegisterHooks(rawCore, func(entry zapcore.Entry) error {
+		if entry.Level == zapcore.FatalLevel {
```
Comment (wklken) [inline src/core-api/pkg/logging/init.go:82]: 只有fatal 才上报?

**Review by wklken — APPROVED** [review]

**Review by wklken — APPROVED** [review]

File: src/core-api/pkg/logging/init.go [hunk 3]
```diff
@@ -49,15 +52,41 @@ func newZapSugarLogger() *zap.SugaredLogger {
 }
 
 // InitLogger ...
-func InitLogger(logger *config.Logger) {
+func InitLogger(config *config.Config) {
 	loggerInitOnce.Do(func() {
-		defaultLogger = newZapJSONLogger(&logger.Default).Sugar()
+		options := make([]zap.Option, 0, 3)
+		if len(strings.TrimSpace(config.Sentry.DSN)) != 0 {
```
Comment (alex-smile) [inline src/core-api/pkg/logging/init.go:58]: config.Sentry.DSN 直接返回 TimeSpace 后的结果

File: src/core-api/cmd/init.go [hunk 4]
```diff
@@ -67,7 +69,7 @@ func initDatabase() {
 }
 
 func initSentry() {
-	if globalConfig.Sentry.Enable {
+	if len(strings.TrimSpace(globalConfig.Sentry.DSN)) != 0 {
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/core-api/cmd/init.go]: done

**Pull request merged**